<launch>
  <arg name="output" default="log"/>
  <arg name="standalone" default="false"/>

  <!-- World setup -->
  <include if="$(arg standalone)" file="$(find agimus_demos)/launch/tiago_deburring_world_setup.launch">
  </include>

  <!-- Setup vision -->
  <group ns="vision">
    <node name="camera_to_xtion" pkg="tf2_ros" type="static_transform_publisher"
    args="-0.016 -0.015 0.064 1.5707963 -1.5707963 0.0 xtion_rgb_optical_frame camera_link"/>
    <node pkg="agimus_vision" type="tracker_object"
        output="$(arg output)"
        name="tracker"
        >
        <param name="imageTopic"      value="/camera/color/image_raw" />
        <param name="cameraInfoTopic" value="/camera/color/camera_info" />
        <param name="depthImageTopic" value="/camera/aligned_depth_to_color/image_raw" />
        <param name="debugDisplay"    value="false" />
        <param name="broadcastTopic"  value="true" />
        <param name="broadcastTf"     value="true" />
        <param name="broadcastTfPostfix" value="_tf" />
        <param name="objectType" value="apriltag" />
        <param name="cameraFrame" value="camera_color_optical_frame" />
        <param name="depthScale"      value="0.1"/>  
    </node>

    <!-- Part -->
   <node pkg="agimus_demos" type="group_of_tags.py" name="part_tags" args="
        --child-frame-format 'part/tag36_11_{:0>5d}'
        --measurement-parent-frame camera_color_optical_frame
        --group-frame part/base_link
        --no-description
        --size 0.0845
        --tag 1
        --size 0.0845
        --tag 6
        --size 0.0845
        --tag 15
        --size 0.0845
        --tag 13
        --size 0.0845
        --tag 14
        --size 0.0845
        --tag 5
        --size 0.06
        --tag 100
        --size 0.06
        --tag 101
        --size 0.06
        --tag 102
        "
        output="$(arg output)"
        />

    <!-- Driller -->
       <node pkg="agimus_demos" type="group_of_tags.py" name="driller_tags" args="
        --child-frame-format 'driller/tag36_11_{:0>5d}'
        --measurement-parent-frame camera_color_optical_frame
        --group-frame driller/base_link
        --no-description
        --size 0.04
        --tag 230
        --size 0.04
        --tag 232
        --size 0.04
        --tag 23
        "
        output="$(arg output)"
        />
   
  </group>
</launch>
